### PR TITLE
build: use whiskers to generate themes

### PR DESCRIPTION
**Pull request by tranzystorekk**
This PR lines up the halloy theme with the new standard of the catppuccin ecosystem: use the `whiskers` tool to generate themes from a template.

Adds a just script like other similar proposed changes, e.g. https://github.com/catppuccin/openbox/pull/2

Also changes the generated format to toml, as required by halloy 2024.5 and up, although I guess we could also generate legacy yamls if we want more backward compat?

### CODE REVIEW

**Comment by tranzystorekk** [comment]
https://github.com/catppuccin/halloy/pull/1#issuecomment-2110824114 @uncenter I took a stab at contributing a whiskers change, hope it's okay

File: README.md [hunk 1]
```diff
@@ -36,14 +36,13 @@
 
 ## Usage
 
-1. Clone this repository locally
-2. Copy the desired `<theme>.yaml` file into your themes directory:  
+1. Download your flavor of choice from [`themes/`](./themes/) to halloy's themes directory:
     - Linux: `$HOME/.config/halloy/themes`
     - macOS: `$HOME/Library/Application Support/halloy/themes`
     - Windows: `{FOLDERID_RoamingAppData}\halloy\themes`
```
Comment (uncenter) [inline README.md]:
```suggestion
1. Download your flavor of choice from [`themes/`](./themes/) to the `themes/` directory inside [Halloy's configuration directory](https://halloy.squidowl.org/configuration/index.html).
```

File: README.md [hunk 2]
```diff
@@ -36,14 +36,13 @@
...
-4. Update the `theme:` field of your `config.yaml` to the name of the theme without the `.yaml` file extension:
-    - `theme: "catppuccin-frappe"`
-5. Restart Halloy
+2. Update the `theme` field of your `config.toml` to the name of the theme without the `.toml` file extension:
+    - `theme = "catppuccin-frappe"`
```
Comment (uncenter) [inline README.md]:
```suggestion
2. Update the `theme` field of your `config.toml` to `catppuccin-<flavor>`, where `<flavor>` is your chosen flavor. For example, `theme = "catppuccin-frappe"`.
```

File: README.md [hunk 3]
```diff
@@ -36,14 +36,13 @@
...
+3. Restart Halloy
```
Comment (uncenter) [inline README.md]:
```suggestion
3. Restart Halloy to see the theme applied.
```

File: halloy.tera [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+whiskers:
+  version: "2.4.0"
+  matrix:
+    - flavor
+  filename: "themes/catppuccin-{{flavor.identifier}}.toml"
+---
+name = "catppuccin-{{flavor.identifier}}"
```
Comment (uncenter) [inline halloy.tera]:
```suggestion
  filename: "themes/catppuccin-{{ flavor.identifier }}.toml"
---
name = "catppuccin-{{ flavor.identifier }}"
```

More readable with better whitespace :)

File: halloy.tera [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+[palette]
+background = "#{{flavor.colors.base.hex}}"
+text = "#{{flavor.colors.text.hex}}"
+action = "#{{flavor.colors.rosewater.hex}}"
+accent = "#{{flavor.colors.subtext1.hex}}"
+alert = "#{{flavor.colors.yellow.hex}}"
+error = "#{{flavor.colors.red.hex}}"
+info = "#{{flavor.colors.blue.hex}}"
+success = "#{{flavor.colors.green.hex}}"
```
Comment (uncenter) [inline halloy.tera]:
```suggestion
background = "#{{ base.hex }}"
text = "#{{ text.hex }}"
action = "#{{ rosewater.hex }}"
accent = "#{{ subtext1.hex }}"
alert = "#{{ yellow.hex }}"
error = "#{{ red.hex }}"
info = "#{{ blue.hex }}"
success = "#{{ green.hex }}"
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Review by uncenter — APPROVED** [review]

**Review by sgoudham — APPROVED** [review]
@Daeraxa Are you happy if I go ahead and merge this? It would supercede the changes made in #1.

I know you're said pretty busy so no pressure to reply as soon as you see this notification. This PR looks good to me so I'll most likely merge in the coming days if you don't get to it before me :+1:

**Comment by tranzystorekk** [comment]
Halloy has new pending theme config changes, and quite substantial ones: https://unstable.halloy.squidowl.org/configuration/themes/

It'll probably take me a bit to find correct palette maappings, so we should hold off merging for now.

**Comment by sgoudham** [comment]
I think it's good to merge as is now so that it's easier to generate the new mappings later on with the new configuration. Would be good to raise an issue for that. @tranzystorekk

**Comment by sgoudham** [comment]
Making an "executive" decision to merge this and create a tracking issue on the repository for the configuration to be updated with the new mappings which someone can pick up.

Thanks for your work! @tranzystorekk

**Pull request merged**